Sage/carbon
# fix(file-input): ensure that long file names are truncated correctly - FE-6685

### Proposed behaviour

![Screenshot 2024-07-16 at 14 10 22](https://github.com/user-attachments/assets/0a3921ad-26a8-41e5-a34a-091445026b06)

### Current behaviour

![Screenshot 2024-07-16 at 14 10 48](https://github.com/user-attachments/assets/9e3b5a65-6c1f-4d01-bfe6-e9043fb27ac5)

### Checklist




#### QA


### Additional context

This bug can only been seen on Chrome and Firefox. 

### Testing instructions
On all browsers we test on including Windows too: 

- Check the story `file-input-test--long-filename-status` located in the `Test` folder for `FileInput`. The long file name should be truncated. 

- You could also check the `upload-status-alternative` example and upload files of varying lengths of file name to check this also works as expected. 

- Could be worth just regression testing the other examples to make sure nothing else has cropped up because of this change.
